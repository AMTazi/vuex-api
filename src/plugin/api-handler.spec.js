import { mount } from 'vue-test-utils'
import ApiHandler from './api-handler'

describe('Api handler', () => {
  it('call the correct hooks when mounting', () => {
    const dispatch = jest.fn()
    mount(ApiHandler(), { mocks: { $store: { dispatch } }, propsData: { url: 'some/url', keyPath: 'totally-a-keypath' } })
    expect(dispatch.mock.calls).toMatchSnapshot()
  })

  it('should be called after destroy', () => {
    const dispatch = jest.fn()
    const wrapper = mount(ApiHandler({ baseURL: 'https://api.github.com' }), { mocks: { $store: { dispatch } }, propsData: { persistent: false, url: 'some/url', keyPath: 'totally-a-keypath' } })
    wrapper.vm.$destroy()
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})
