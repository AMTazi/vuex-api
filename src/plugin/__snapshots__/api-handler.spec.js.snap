// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api handler call the correct hooks when mounting 1`] = `
Array [
  Array [
    "Request_Api_Call",
    Object {
      "keyPath": "totally-a-keypath",
      "url": "some/url",
    },
  ],
]
`;

exports[`Api handler should be called after destroy 1`] = `
Array [
  Array [
    "Request_Api_Call",
    Object {
      "baseURL": "https://api.github.com",
      "keyPath": "totally-a-keypath",
      "url": "some/url",
    },
  ],
  Array [
    "Clear_Api_Call",
    "totally-a-keypath",
  ],
]
`;
